libs/core/src/lib/select/select.component.html

```diff
@@ -1,95 +1,113 @@
-             role="listbox"
-             #selectControl
-             [ngClass]="state ? 'is-' + state : ''"
-             [class.is-active]="isOpen"
-             [class.is-expanded]="isOpen"
-             [class.is-disabled]="disabled"
-             [class.is-readonly]="readonly"
-             [attr.tabindex]="disabled ? -1 : 0"
-             [attr.id]="controlId"
-             [attr.aria-active]="isOpen"
-             [attr.aria-expanded]="isOpen"
-             [attr.aria-disabled]="disabled"
-             [attr.aria-readonly]="readonly"
-             [attr.aria-controls]="controlId + '-list-box'"
-             [attr.aria-haspopup]="isInteractive"
-             [attr.aria-label]="ariaLabel"
-             [attr.aria-labelledby]="ariaLabelledBy"
-             [attr.aria-activedescendant]="selected?.id || null">
-
-            <div class="fd-select__content">
-                <ng-container
-                    [ngTemplateOutlet]="controlTemplate ? controlTemplate : defaultSelectContent"
-                    [ngTemplateOutletContext]="{$implicit: selectViewValue, selected: selected}">
-                </ng-container>
-
-                <ng-template #defaultSelectContent>
-                    {{ selectViewValue }}
-                </ng-template>
-            </div>
+    <div
+        class="fd-select__control"
+        role="listbox"
+        #selectControl
+        [ngClass]="state ? 'is-' + state : ''"
+        [class.is-active]="panelOpen"
+        [class.is-expanded]="panelOpen"
+        [class.is-disabled]="disabled"
+        [class.is-readonly]="readonly"
+        [attr.tabindex]="tabIndex"
+        [attr.id]="controlId"
+        [attr.aria-active]="panelOpen"
+        [attr.aria-expanded]="panelOpen"
+        [attr.aria-disabled]="disabled"
+        [attr.aria-readonly]="readonly"
+        [attr.aria-controls]="controlId + '-list-box'"
+        [attr.aria-haspopup]="!(this.readonly || this.disabled)"
+        [attr.aria-label]="ariaLabel"
+        [attr.aria-labelledby]="ariaLabelledBy"
+        [attr.aria-activedescendant]="getAriaActiveDescendant()"
+        (focus)="onFocus()"
+        (blur)="onBlur()"
+    >
+        <div class="fd-select__content">
+            <ng-container
+                [ngTemplateOutlet]="controlTemplate ? controlTemplate : defaultSelectContent"
+                [ngTemplateOutletContext]="{ $implicit: triggerValue, selected: selected }"
+            >
+            </ng-container>
 
-            <button fd-button
-                    tabindex="-1"
-                    class="fd-button--transparent fd-select__button"
-                    *ngIf="!readonly"
-                    [disabled]="disabled"
-                    [glyph]="glyph">
-            </button>
+            <ng-template #defaultSelectContent>
+                {{ triggerValue }}
+            </ng-template>
         </div>
+        <button
+            fd-button
+            tabindex="-1"
+            class="fd-button--transparent fd-select__button"
```

Comment:
Shouldn't we have button input to deal with `--transparent`?

Reply:
@JKMarkowski  I will resolve this one as I cant say much to it. This something you guys can address later if you decide if the button trigger needs different styling